src/main/kotlin/com/github/rushyverse/pvpbox/map/MapImage.kt

```diff
@@ -0,0 +1,97 @@
+ * @property packets The packets list to send to new players.
+ * @constructor
+ */
+class MapImage(
+    private val resourceImageName: String,
+    private val widthBlocks: Int,
+    private val heightBlocks: Int,
+) {
+
+    companion object {
+        /**
+         * Corresponds to the default resolution used to build image as packets.
+         */
+        const val DEFAULT_RESOLUTION = 128;
+
+        /**
+         * Create an item frame on which the image will be displayed
+         * @param instance The instance where you want to create the frame.
+         * @param maximum The position of the frame.
+         */
+        fun createItemFrame(instance: Instance, maximum: Pos) {
+            val maxX = maximum.blockX()
+            val maxY = maximum.blockY()
+            val z = maximum.blockZ()
+            val yaw = maximum.yaw
+            for (i in 0..14) {
+                val x = maxX - i % 5
+                val y = maxY - i / 5
+                val itemFrame = Entity(EntityType.ITEM_FRAME)
+                val meta = itemFrame.entityMeta as ItemFrameMeta
+                itemFrame.setInstance(instance, Pos(x.toDouble(), y.toDouble(), z.toDouble(), yaw, 0f))
+                meta.setNotifyAboutChanges(false)
+                meta.orientation = ItemFrameMeta.Orientation.NORTH
+                meta.isInvisible = true
+                meta.item = ItemStack.builder(Material.FILLED_MAP)
+                    .meta(MapMeta::class.java) { builder: MapMeta.Builder -> builder.mapId(i) }
+                    .build()
+                meta.setNotifyAboutChanges(true)
+            }
+        }
+
+        /**
+         * Creates packets from the image.
+         * @param framebuffer The frame buffer to convert as packets.
+         * @return The list of packets.
+         */
+        private fun mapPackets(framebuffer: LargeGraphics2DFramebuffer): Array<SendablePacket?> {
+            return Array(15) {
+                val x = it % 5
+                val y = it / 5
+                framebuffer.createSubView(x shl 7, y shl 7).preparePacket(it)
+            }
+        }
+    }
+
+    private var packets: Array<SendablePacket?>? = null
+
+    /**
+     * Get the packets of the current Map Image.
+     * This function generates and stores packets if they are not created.
+     * @return The list of packets.
+     */
+    fun packets(): Array<SendablePacket?>? {
+        return if (packets != null)
+            packets
+        else {
+            val framebuffer =
+                LargeGraphics2DFramebuffer(widthBlocks * DEFAULT_RESOLUTION, heightBlocks * DEFAULT_RESOLUTION)
+            javaClass.getResourceAsStream("/$resourceImageName")!!.buffered().use {
+                val image = ImageIO.read(it)
+                framebuffer.renderer.drawRenderedImage(image, AffineTransform.getScaleInstance(1.0, 1.0))
+                packets = mapPackets(framebuffer)
+            }
+            packets
+        }
+    }
+}
```

Comment:
For a lazy initialization, use `lazy` keyword

```suggestion
    private var packets: Array<SendablePacket?> by lazy {
                val framebuffer =
                LargeGraphics2DFramebuffer(widthBlocks * DEFAULT_RESOLUTION, heightBlocks * DEFAULT_RESOLUTION)
            javaClass.getResourceAsStream("/$resourceImageName")!!.buffered().use {
                val image = ImageIO.read(it)
                framebuffer.renderer.drawRenderedImage(image, AffineTransform.getScaleInstance(1.0, 1.0))
                mapPackets(framebuffer)
            }
    }
```